src/shared/ui/Chip.tsx

```diff
@@ -0,0 +1,61 @@
+import { View } from 'react-native';
+
+import { styled } from '@tamagui/core';
+import { SvgProps } from 'react-native-svg';
+
+import { Text } from './Text';
+import { colors } from '../lib/constants/colors';
+
+const getFillForVariant = (
+  variant: React.ComponentProps<typeof StyledChip>['variant'],
+) => {
+  switch (variant) {
+    case 'warning':
+      return '#D9730D';
+    case 'primary':
+    default:
+      return colors.alertInfoIcon;
+  }
+};
+
+const StyledChip = styled(View, {
+  alignItems: 'center',
+  borderRadius: 4,
+  flexDirection: 'row',
+  gap: 4,
+  name: 'Chip',
+  px: 8,
+  py: 4,
```

Comment:
Nice Chip! Did you base this on an existing Figma component? If so, I'm happy to defer to that, but the only thing I could find was the one from the ML1 [Design System](https://www.figma.com/design/LXl9BAJSwtgWuzkxqOqvX8/ML1-Design-System?node-id=58697-179569&m=dev). There's a few very subtle differences, but I thought I'd offer my suggestions here (but point me to the right direction if there's another mobile design you referenced).

| Before | After |
| - | - |
| ![CleanShot 2024-10-23 at 20 03 10@2x](https://github.com/user-attachments/assets/0ab37141-f0b4-424e-bb63-fe7222ec9ae6) | ![CleanShot 2024-10-23 at 20 03 40@2x](https://github.com/user-attachments/assets/c501f08a-1ecf-42ce-a93e-9df163889275) |

Here, I changed:

```tsx
  borderRadius: 8,
  gap: 8,
  pl: 8,
  pr: 12,
  py: 6,
```

And then below, in the `Chip` component definition, I:
- Adapted `lineHeight` to be proportional to the smaller font size of `12px` (in Figma, line height is 20px but for a 14px font size, i.e. line height is 142% of font size)
- Font weight should be 400
- I added `flex={1}` to prevent wrapping text from sometimes overflowing the bounds of the chip (try on iPhone 15 or iPhone 15 Pro)

```tsx
      <Text fontSize={12} fontWeight="400" lineHeight={17} flex={1}>
```

Reply:
The styles here (padding, borderRadius, colors, etc) were a direct copy of the existing label in the web app — the linked task was to adapt the web UI to mobile.

_That_ label was itself styled as a one-off component, but I couldn't help but notice that it was identical in all respects except for colors to a number of variations throughout the web app. The ones I spotted were also built as one-offs.

I couldn't find anything in Figma or documentation elsewhere that matched the appearance of this pattern in the web app, but it's the pattern as it appeared _there_ that I attempted to formalize _here_. 

I saw the `Chip` you linked to, but it seemed distinct enough (different styles, as you point out), and had a lot more states that weren't reflected in the web-app. Maybe I should have named it something different, but I wasn't trying to replicate _that_ `Chip`, specifically 😅.

I went ahead and updated the styles to mostly match your suggestions, regardless — I think the after is much improved. I kept the horizontal padding equal, though, mostly to avoid inconsistent values if an icon were omitted.